test/unit_tests/profiling/test_profiling.cpp

```diff
@@ -0,0 +1,243 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): David J. Gardner @ LLNL
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * ---------------------------------------------------------------------------*/
+
+#include <chrono>
+#include <cstdio>
+#include <iostream>
+#include <ostream>
+#include <string>
+#include <thread>
+
+#include "sundials/sundials_math.h"
+#include "sundials/sundials_profiler.h"
+#include "sundials/sundials_types.h"
+
+int sleep(SUNProfiler prof, int sec, double* chrono)
+{
+  std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+  int flag = SUNProfiler_Begin(prof, "sleep");
+  if (flag) return flag;
+  std::this_thread::sleep_for(std::chrono::seconds(sec));
+  flag = SUNProfiler_End(prof, "sleep");
+  if (flag) return flag;
+  std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
+  *chrono =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(end - begin).count() *
+    1e-9;
+  return 0;
+}
+
+int print_timings(SUNProfiler prof)
+{
+  // Output timing in default (table) format
+  int flag = SUNProfiler_Print(prof, stdout);
+  if (flag)
+  {
+    std::cerr << ">>> FAILURE: "
+              << "SUNProfiler_Print returned " << flag << "\n";
+    return 1;
+  }
+
+  flag = SUNProfiler_Print(prof, stdout);
+  if (flag)
+  {
+    std::cerr << ">>> FAILURE: "
+              << "SUNProfiler_Print returned " << flag << "\n";
+    return 1;
+  }
```

Comment:
Remove duplicate blocks until we get around to adding a `SetFormat` option for output
```suggestion
```